test/e2e/heartbeat_test.go

```diff
@@ -0,0 +1,271 @@
+				return err
+			})
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+		}
+
+		var (
+			managedCluster  *clusterv1.ManagedCluster
+			managedClusters = clusterClient.ClusterV1().ManagedClusters()
+		)
+
+		ginkgo.By(fmt.Sprintf("Waiting for ManagedCluster %q to exist", clusterName))
+		err = wait.Poll(1*time.Second, 90*time.Second, func() (bool, error) {
+			var err error
+			managedCluster, err = managedClusters.Get(context.TODO(), clusterName, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+			return true, nil
+		})
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+		gomega.Expect(managedCluster.Spec.HubAcceptsClient).To(gomega.Equal(false))
+
+		ginkgo.By(fmt.Sprintf("Accepting ManagedCluster %q", clusterName))
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			var err error
+			managedCluster, err = managedClusters.Get(context.TODO(), managedCluster.Name, metav1.GetOptions{})
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+			managedCluster.Spec.HubAcceptsClient = true
+			managedCluster.Spec.LeaseDurationSeconds = 5
+			managedCluster, err = managedClusters.Update(context.TODO(), managedCluster, metav1.UpdateOptions{})
+			return err
+		})
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+		ginkgo.By("Waiting for ManagedCluster is accepted")
+		err = wait.Poll(1*time.Second, 90*time.Second, func() (bool, error) {
+			var err error
+			managedCluster, err := managedClusters.Get(context.TODO(), clusterName, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+
+			condition := helpers.FindManagedClusterCondition(managedCluster.Status.Conditions, clusterv1.ManagedClusterConditionHubAccepted)
+			return helpers.IsConditionTrue(condition), nil
+		})
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+		ginkgo.By("Make sure ManagedCluster is available")
+		err = wait.Poll(1*time.Second, 90*time.Second, func() (bool, error) {
+			managedCluster, err := managedClusters.Get(context.TODO(), clusterName, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+
+			condition := helpers.FindManagedClusterCondition(managedCluster.Status.Conditions, clusterv1.ManagedClusterConditionAvailable)
+			return helpers.IsConditionTrue(condition), nil
+		})
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+		leaseName := fmt.Sprintf("cluster-lease-%s", clusterName)
+		ginkgo.By(fmt.Sprintf("Make sure ManagedCluster lease %q exists", leaseName))
+		var lastRenewTime *metav1.MicroTime
+		err = wait.Poll(1*time.Second, 30*time.Second, func() (bool, error) {
+			lease, err := hubClient.CoordinationV1().Leases(clusterName).Get(context.TODO(), leaseName, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+			lastRenewTime = lease.Spec.RenewTime
+			return true, nil
+		})
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+		ginkgo.By(fmt.Sprintf("Make sure ManagedCluster lease %q is updated", leaseName))
```

Comment:
Why not just tack this onto the loopback test? That test can just be extended with the lease logic to become more thorough. WDYT?

Reply:
yes, it make sense, I will add this to loopback test, the loopback will test all positive cases